### PR TITLE
Simplify the data preprocessing code for the sentiment analysis script

### PR DESCRIPTION
**Pull request by leezu**
## Description ##
Simplify the data preprocessing code for the sentiment analysis. There is no point in using `multiprocessing` for such small datasets.

### Essentials ###
- [X] Changes are complete (i.e. I finished coding on this PR)
- [X] All changes have test coverage
- [X] Code is well-documented

### Changes ###
- [X] Simplify the data preprocessing code for the sentiment analysis script

### CODE REVIEW

File: scripts/sentiment_analysis/process_data.py [hunk 1]
```diff
@@ -98,30 +101,24 @@ def _build_vocab(data_name, train_dataset, test_dataset):
...
-    return float(len(x[0]))
 
 def _preprocess_dataset(dataset, vocab, max_len):
     start = time.time()
-    pool = mp.Pool(8)
-    partial_work = partial(_preprocess, vocab=vocab, max_len=max_len)
-    dataset = pool.map(partial_work, dataset)
-    lengths = gluon.data.SimpleDataset(pool.map(_get_length, dataset))
+    dataset = [_preprocess(d, vocab=vocab, max_len=max_len) for d in dataset]
+    lengths = gluon.data.SimpleDataset([len(d[0]) for d in dataset])
```
Comment (astonzhang) [inline scripts/sentiment_analysis/process_data.py]: Why do you want to remove pool?

P.S. pls fix sanity check errors.

**Comment by mli** [comment]
Job PR-462/2 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-462/2/index.html

File: scripts/sentiment_analysis/process_data.py [hunk 2]
```diff
@@ -86,7 +86,7 @@ def _build_vocab(data_name, train_dataset, test_dataset):
         test_dataset[i][0] = _clean_str(line[0], data_name)
         line = test_dataset[i][0].split()
         max_len = max_len if max_len > len(line) else len(line)
-    all_token.extend(line)
+        all_token.extend(line)
```
Comment (leezu) [inline scripts/sentiment_analysis/process_data.py]: @xiaotinghe could you explain this change? Did I change it by accident or was there a bug before? Thanks for checking the changes!

Comment (xiaotinghe) [reply scripts/sentiment_analysis/process_data.py]: There was a bug before.
I happened to see it.

Comment (astonzhang) [reply scripts/sentiment_analysis/process_data.py]: Will this affect the results in logs?

Comment (leezu) [reply scripts/sentiment_analysis/process_data.py]: @xiaotinghe the CI fails since your change, can you make it pass again?

Comment (szha) [reply scripts/sentiment_analysis/process_data.py]: @leezu there was a problem with nbsphinx version that @eric-haibin-lin fixed just now. Try doing and rebase and see if the issue persists

**Comment by mli** [comment]
Job PR-462/6 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-462/6/index.html

**Pull request merged**